test/counter.go

```diff
@@ -19,6 +19,16 @@ import (
 
 type BaseValue func(*toolchainv1alpha1.ToolchainStatus)
 
+func ToolchainStatus(baseValues ...BaseValue) *v1alpha1.ToolchainStatus {
+	toolchainStatus := &v1alpha1.ToolchainStatus{
+		Status: v1alpha1.ToolchainStatusStatus{},
+	}
+	for _, apply := range baseValues {
+		apply(toolchainStatus)
+	}
+	return toolchainStatus
+}
```

Comment:
TBH, this is very confusing - there is already `NewToolchainStatus` https://github.com/codeready-toolchain/host-operator/blob/master/test/toolchainstatus.go#L65-L76 function accepting different option type 
Also, when you take a look at your changes, sometimes you use:
```
NewToolchainStatus(WithHost(WithMasterUserRecordCount(20)))
``` 
but in other cases
```
ToolchainStatus(MasterUserRecords(2), UserAccountsForCluster(test.MemberClusterName, 2))
```
For me, it was hard to understand the difference and why do we have two similar ways of creating `ToolchainStatus` CR

Reply:
ok, so maybe by merging those 2 funcs it would be clearer?

Reply:
Either that or keeping the original version.

Reply:
One of the goal of this PR is to show that counters are initialized from a `ToolchainStatus` resource (this was hidden within the `initializeCounters` func), so I'd rather merge the 2 funcs and move on, instead of moving backward

Reply:
ok, then feel free to merge them.

Reply:
done in b93458a. Is it better for you like that?

Reply:
Yes, thank you